cellengine/resources/gate.py

```diff
@@ -193,51 +96,177 @@ def create(
+                (after the channel's scale has been applied).
+            x2 (float): The second x coordinate (after the channel's scale
+                has been applied).
+            y1 (float): The first y coordinate (after the channel's scale has
+                been applied).
+            y2 (float): The second y coordinate (after the channel's scale has
+                been applied).
+            label (float): Position of the label. Defaults to the midpoint of the gate.
+            gid (str): Group ID of the gate, used for tailoring. If this is not
+                specified, then a new Group ID will be created. To create a
+                tailored gate, the gid of the global tailored gate must be specified.
+            locked (bool): Prevents modification of the gate via the web interface.
+            tailored_per_file (bool): Whether or not this gate is tailored per FCS file.
+            fcs_file_id (str): ID of FCS file, if tailored per file. Use ``None`` for
+                the global gate in a tailored gate group. If specified, do not
+                specify ``fcs_file``.
+            fcs_file (str): Name of FCS file, if tailored per file. An attempt
+                will be made to find the file by name. If zero or more than one file
+                exists with the name, an error will be thrown. Looking up files
+                by name is slower than using the ID, as this requires
+                additional requests to the server. If specified, do not specify
+                ``fcs_file_id``.
 
-class PolygonGate(Gate):
-    """Basic concrete class for polygon gates"""
+        Returns:
+            A RectangleGate object.
+
+        Example:
+            ```python
+            experiment.create_rectangle_gate(x_channel="FSC-A", y_channel="FSC-W",
+            name="my gate", 12.502, 95.102, 1020, 32021.2)
+            cellengine.Gate.create_rectangle_gate(experiment_id, x_channel="FSC-A",
+            y_channel="FSC-W", name="my gate", x1=12.502, x2=95.102, y1=1020,
+            y2=32021.2, gid=global_gate.gid)
+            ```
+        """
+
+        fcs_file_id = parse_fcs_file_args(
+            experiment_id,
+            tailored_per_file,
+            fcs_file_id,
+            fcs_file,
+        )
 
+        if label == []:
+            label = [numpy.mean([x1, x2]), numpy.mean([y1, y2])]
+        if not gid:
+            gid = generate_id()
+
+        model = {
+            "locked": locked,
+            "label": label,
+            "rectangle": {"x1": x1, "x2": x2, "y1": y1, "y2": y2},
+        }
+
+        return Gate(
+            id=None,
+            experiment_id=experiment_id,
+            fcs_file_id=fcs_file_id,
+            gid=gid,
+            tailored_per_file=tailored_per_file,
+            type="RectangleGate",
+            x_channel=x_channel,
+            model=model,
+            y_channel=y_channel,
+            name=name,
+        )  # type: ignore
```

Comment:
Re: the `# type: ignore`, the returned object is actually an instance of `Gate` and not a `RectangleGate`, right? For now that probably doesn't matter, but since Python doesn't really have a good flow cytometry library like R does, we might implement `apply` methods for each subclass, in which case it would need to be the correct type.

Reply:
I'll have to think about this some more :thinking:

Reply:
Maybe I'll be able to resolve this following the discussion here https://github.com/primitybio/cellengine-python-toolkit/pull/136#discussion_r796215409